NetKet/Operator/local_operator.hpp

```diff
@@ -293,6 +297,17 @@ class LocalOperator : public AbstractOperator {
     return LocalOperator(lhs.GetHilbert(), mat, sites);
   }
 
+  void operator=(const LocalOperator &rhs) {
+    assert(rhs.hilbert_.LocalStates().size() == hilbert_.LocalStates().size());
```

Comment:
OK, but maybe we can also check the other properties are equal? 
Would it work to do 
```c++
assert(rhs.hilbert_==hilbert_);
```

? Maybe not

Reply:
Definitely not the last `assert`! We can set up a comparison operator in the abstract `hilbert` to checkout out that the properties are equal... but that's for another PR, no worries.